server.go

```diff
@@ -54,13 +54,41 @@ func NewServer(etcd *etcdv3.Client, cfg ServerCfg) (*Server, error) {
 	if etcd == nil {
 		return nil, ErrNilEtcd
 	}
-	return &Server{
+
+	// Create a registry client, through which other
+	// entities like peers, actors, and mailboxes
+	// will be discovered.
+	r, err := registry.New(etcd)
```

Comment:
More details exist in the individual commit, but basically the registry needs to be created in NewServer, not Serve, as things call Serve at the same time that modification to the registry is happening. This was detected by the race detector while trying to diagnose the above.

The ctx and cancel move is the exact same thing, but raced at a slightly different location.